tests/test_stomp.py

```diff
@@ -63,3 +63,87 @@ def test_stomp_A_B_join(T_A, T_B):
     utils.replace_inf(left)
     utils.replace_inf(right)
     npt.assert_almost_equal(left, right)
+
+
+def test_stomp_nan_self_join_beginning():
+    m = 3
+    T = np.array([np.nan, 1, 0, 0, 1, 0, 0])
+
+    zone = int(np.ceil(m / 4))
+    left = np.array(
+        [
+            utils.naive_mass(Q, T, m, i, zone, ignore_trivial=True)
+            for i, Q in enumerate(core.rolling_window(T, m))
+        ],
+        dtype=object,
+    )
+
+    right = stomp(T, m, ignore_trivial=True)
+
+    utils.replace_inf(left)
+    utils.replace_inf(right)
+    npt.assert_almost_equal(left[:, 0], right[:, 0])
+
+
+def test_stomp_inf_self_join_beginning():
+    m = 3
+    T = np.array([np.inf, 1, 0, 0, 1, 0, 0])
```

Comment:
Is there a reason why the rest of the sequence is `[1, 0, 0, 1, 0, 0]`? Could we use the `test_data` and replace the first element with `np.inf`?